docs-v2/customize/guides/advanced/handle-large-datasets.mdx

```diff
@@ -4,6 +4,123 @@ sidebarTitle: 'Handle large datasets'
-</Info>
\ No newline at end of file
+A common type of integration is to let users view & interact with data from external systems. If they only view the data, you need to maintain a 1-way sync of the data (from the external system to your app); if they can edit it, you need to maintain a 2-way sync. 
+
+In both cases, you need to transfer data from the external system to your app, either on a schedule or in real-time. In Nango, this is achieved with a [sync](/understand/concepts/syncs). For a 2-way sync, you can write back to the external system using [actions](/understand/concepts/actions).
+
+# Context
+
+When syncing data in Nango, there are actually two syncing processes happening: 
+- Nango syncing with the external system, managed by [integration scripts](/understand/concepts/scripts)
+- Your app syncing with Nango, managed by your app's code using the Nango API
+
+Your app syncs with Nango in the same way regardless of the volume of the dataset ([guide](/integrate/guides/sync-data-from-an-api)). Nango maintains a cache of the synced data, and the Nango API allows you to fetch only the incremental changes for your convenience. 
+
+Below, we explore various approaches on how to sync data from the external system to Nango, even for large datasets.
+
+# Full refresh syncing (small datasets only)
+
+For small datasets (e.g., a list of Slack users for organizations with less than 100 employees), you can instruct Nango to periodically poll for the entire dataset. This method is known as a ["full refresh" sync](/understand/concepts/syncs#full-refresh-syncs).
+
+Although the dataset is overridden with each sync execution, this is abstracted away from your app's logic, allowing it to still benefit from fetching only incremental changes from the Nango API. 
+
+As datasets increase in size, full refresh syncs become unscalable, taking longer to run, triggering rate limiting from external APIs, and consuming more computing and memory resources.
+
+Here is an example of a sync script which refreshes the entire list of Slack user on each execution:
+
+```ts
+export default async function fetchData(nango: NangoSync) {
+    // ...
+    // Paginate API requests.
+    while (nextPage) {
+        const res = getUsersFromSlackAPI(nextPage);
+        // ...
+        // Save Slack users.
+        await nango.batchSave(mapUsers(res.data.members), 'User');
+    }
+}
+```
+
+# Incremental syncing
+
+The preferred method for syncing larger datasets is to fetch only the incremental changes from the external API. This method is known as an ["incremental" sync](https://docs.nango.dev/understand/concepts/syncs#incremental-syncs). 
+
+Nango automatically stores the timestamp of the last sync execution start. You can use this timestamp to instruct the external API to send only the changes since that date. That way, you only receive and persist the modified records in the Nango cache.
```

Comment:
I would add a mention to `lastSyncDate` prop. It's the key element to achieve incremental syncing and it's not entirely clear how to achieve that by just reading

Reply:
Agreed, I'll update!